packages/miniflare/src/workers/core/entry.worker.ts

```diff
@@ -351,6 +355,156 @@ async function handleScheduled(
 }
 
+function createForwardableEmailMessage(
+	from: string,
+	to: string,
+	raw: ReadableStream<Uint8Array<ArrayBufferLike>>,
+	headers: Headers,
+	setReject: (reason: string) => void,
+	forward: (rcptTo: string, headers?: Headers) => Promise<void>,
+	reply: (message: EmailMessage) => Promise<void>
+) {
+	return {
+		from,
+		to,
+		raw,
+		headers: headers,
+		rawSize: Number(headers.get("Content-Length")),
+		setReject,
+		forward,
+		reply,
+	} satisfies ForwardableEmailMessage;
+}
+
+async function handleEmail(
+	params: URLSearchParams,
+	request: Request,
+	service: Fetcher,
+	env: Env,
+	_ctx: ExecutionContext
+): Promise<Response> {
+	// `from` and `to` are the equivalent ones to the SMTP RCPT commands (not the ones in the email message itself)
+	const from = params.get("from") ?? "";
+	const to = params.get("to") ?? "";
+	// clone request as we need the body to be cloned to forward it to the inside of the worker
+	const clonedRequest = request.clone();
+	if (!request.body || !clonedRequest.body) {
+		return new Response("Include an email in the body", { status: 400 });
+	}
+
+	// Get request blob as we need its size
+	const incomingEmailRaw = await (await request.blob()).bytes();
+
+	// Email worker limit is 25 MiB total
+	if (incomingEmailRaw.byteLength > 25 * 1024 * 1024) {
+		return new Response("Email is bigger than current limit - 25MiB.", {
+			status: 400,
+		});
+	}
+
+	let parsedIncomingEmail: Email;
+	try {
+		parsedIncomingEmail = await PostalMime.parse(incomingEmailRaw);
+	} catch (e) {
+		const error = e as Error;
+		return new Response(
+			`Email could not be parsed: ${error.name}: ${error.message}`,
+			{ status: 400 }
+		);
+	}
+
+	if (parsedIncomingEmail.messageId === undefined) {
+		return new Response(
+			"Email could not be parsed: invalid or no message id provided",
+			{ status: 400 }
+		);
+	}
+
+	if (from !== parsedIncomingEmail.from.address) {
+		console.log(
+			`${yellow("Provided envelope from address doesn't match message From header")} - Envelope From: ${from}; Header From: ${parsedIncomingEmail.from.address}`
+		);
+	}
+
+	if (!parsedIncomingEmail.to?.map((addr) => addr.address).includes(to)) {
+		console.log(
+			`${yellow("Provided envelope to address doesn't match any message To address")} - Envelope To: ${to}; Header To: ${parsedIncomingEmail.to?.map((addr) => addr.address)}`
+		);
+	}
+
+	let incomingEmailHeaders: Headers;
+	try {
+		incomingEmailHeaders = new Headers(
+			parsedIncomingEmail.headers.map((header) => [header.key, header.value])
+		);
+	} catch (e) {
+		const error = e as Error;
+		return new Response(
+			`Email could not be parsed: ${error.name}: ${error.message}`,
+			{ status: 400 }
+		);
+	}
+
+	let maybeClientError: string | undefined = undefined;
+
+	// @ts-expect-error .email is not in the fetcher but it's a valid RPC call.
+	await service.email(
+		// FIXME(lduarte): I have a feeling that because we are using RPC to do this, from + to + headers + readableStream can only take
```

Comment:
are we tracking this `fixme` item? Also, can that fix land in a followup PR? Is that also part of the dev week deliverable? Just trying to understand

Reply:
I'm going to add a warning about this limitation for users. It's not part of the dev week deliverable, and might not actually be solvable without runtime work.